### PR TITLE
osd: drop the unused request_redirect_t::osd_instructions

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

File: src/osd/osd_types.cc [hunk 1]
```diff
@@ -248,7 +248,8 @@ void request_redirect_t::encode(bufferlist& bl) const
   ENCODE_START(1, 1, bl);
   encode(redirect_locator, bl);
   encode(redirect_object, bl);
-  encode(osd_instructions, bl);
+  bufferlist legacy_osd_inst;
+  encode(legacy_osd_inst, bl);
```
Comment (liewegas) [inline src/osd/osd_types.cc]: even better is to ```encode((uint32_t)0, bl)```

File: src/osd/osd_types.cc [hunk 2]
```diff
@@ -257,7 +258,8 @@ void request_redirect_t::decode(bufferlist::const_iterator& bl)
   DECODE_START(1, bl);
   decode(redirect_locator, bl);
   decode(redirect_object, bl);
-  decode(osd_instructions, bl);
+  bufferlist legacy_osd_inst;
+  decode(legacy_osd_inst, bl);
```
Comment (liewegas) [inline src/osd/osd_types.cc]: same here.. decode a uint32_t and if (len) bl.advance(len)

File: src/osd/osd_types.cc (hunk 1 above)
Comment (rzarzynski) [reply src/osd/osd_types.cc]: Cool, just the size of empty `bufferlist`. Thanks!

**Comment by rzarzynski** [comment]
jenkins retest this please (because `unittest_seastar_messenger` on ARM).

**Comment by rzarzynski** [comment]
Looks we've got an issue with the `make check bot` for ARM64:
```
./install-deps.sh: line 383: /home/jenkins-build/build/workspace/ceph-pull-requests-arm64/install-deps-python2.7_tmp/bin/virtualenv: No such file or directory
```

**Comment by rzarzynski** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-11-14_07:46:43-rados-wip-kefu-testing-2018-11-14-1049-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-11-14_23:26:43-rados-wip-kefu-testing-2018-11-15-0204-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-11-15_04:23:30-rados-wip-kefu-testing-2018-11-15-0204-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-11-15_05:33:10-rados-wip-kefu-testing-2018-11-15-0204-distro-basic-mira/

the test failures are fixed by
- #25106
- #25108

**Pull request merged**